packages/headless/src/features/facets/facet-search-set/specific/specific-facet-search-set-slice.ts

```diff
@@ -33,6 +39,12 @@ export const specificFacetSearchSetReducer = createReducer(
       })
       .addCase(executeFacetSearch.fulfilled, (state, action) => {
         handleFacetSearchFulfilled(state, action.payload);
+      })
+      .addCase(clearFacetSearch, (state, {payload}) => {
+        handleFacetSearchClear(state, payload, buildEmptyResponse);
+      })
+      .addCase(executeSearch.fulfilled, (state) => {
```

Comment:
Why is this now needed ?

Reply:
This addition is not directly related to the `clear` functionality, but to manage clearing stale facet search results when other parameters in the page. I suggested having headless manage this to not have devs need to manage the state themselves. [More context available here](https://github.com/coveo/ui-kit/pull/736#discussion_r619331354).